jina/helper.py

```diff
@@ -693,12 +696,16 @@ def run(self):
         return asyncio.run(func(*args, **kwargs))
 
 
-def check_keys_exist(keys_to_check, existing_keys):
-    missed = []
-    for k in keys_to_check:
-        if k not in existing_keys:
-            missed.append(k)
-    return missed
+def filter_keys_nonexistent(keys: Iterator, keys_existing: Iterator, check_path: str, logger: 'JinaLogger'):
+    keys = set(keys)
+    missed = set()
+    for key in keys:
+        if key not in keys_existing:
+            missed.add(key)
+    if missed:
+        logger.warning(f'Key(s) {missed} were not found in {check_path}. Continuing anyway...')
+        keys -= missed
+    return keys
```

Comment:
`keys` could be a generator instead. Therefore we wouldn't have to load all keys in memory if used somewhere else. In our case we have the keys in memory anyways.

Reply:
Well they are already loaded in memory, for performing the calculation above